pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1043 @@
+
+			// coreos imagestream is base, it's special, it needs to pull that image
+			if imageStreamName == pbr.ImageStream.CoreOS {
+
+				// This gets the "rhel-coreos" image from our release out of our controller config
+				controllerConfig, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+				if err != nil {
+					return fmt.Errorf("could not get ControllerConfig %w", err)
+				}
+
+				newImageStream.Spec = imagev1.ImageStreamSpec{
+					LookupPolicy:          imagev1.ImageLookupPolicy{Local: false},
+					DockerImageRepository: "",
+					Tags: []imagev1.TagReference{
+						{
+							Name: "latest",
+							From: &corev1.ObjectReference{
+								Kind: "DockerImage",
+								Name: controllerConfig.Spec.BaseOperatingSystemContainer,
+							},
+						},
+					},
+				}
+
+			}
+
+			// TODO(jkyros): your data structure for this is clearly inelegant, fix it
+			if imageStreamName == pbr.ImageStream.CoreOS || imageStreamName == pbr.ImageStream.RenderedConfig {
+				newImageStream.Annotations["machineconfig.openshift.io/buildconfig"] = pbr.BuildConfig.Content.Name
+			}
+			if imageStreamName == pbr.ImageStream.Content {
+				newImageStream.Annotations["machineconfig.openshift.io/buildconfig"] = pbr.BuildConfig.CustomContent.Name
+			}
+
+			// It didn't exist, put the imagestream in the cluster
+			_, err = ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Create(context.TODO(), newImageStream, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("Attempted to create ImageStream %s but failed: %w", newImageStream, err)
+			}
+			glog.Infof("Created image stream %s", imageStreamName)
+		} else if err != nil {
+			// I don't know if it existed or not, I couldn't get it
+			return fmt.Errorf("Failed to retrieve ImageStream %s: %w", imageStreamName, err)
+		}
+
+	}
+
+	var shouldBeBuilding bool
+	var areWeBuilding bool
+
+	// Figure out which imagestream the pool is deploying from
+	poolImageStreamName, err := ctrlcommon.GetPoolImageStream(pool)
+	if err != nil {
+		return err
+	}
+
+	// Get the actual image stream object for that imagestream
+	poolImageStream, err := ctrl.isLister.ImageStreams(ctrlcommon.MCONamespace).Get(poolImageStreamName)
+	if err != nil {
+		return err
+	}
+
+	// Get the most recent image from that stream if it exists
+	// TODO(jkyros): this can be nil
+	mostRecentPoolImage := ctrl.getMostRecentImageTagForImageStream(poolImageStream, "latest")
+
+	if mostRecentPoolImage != nil {
+		for _, imageStreamName := range ensureImageStreams {
+			var checkImageStream *imagev1.ImageStream
+			// Check to see if we have the imagestream already
+			checkImageStream, err = ctrl.isLister.ImageStreams(ctrlcommon.MCONamespace).Get(imageStreamName)
+			if err != nil {
+				return err
+			}
+
+			mostRecentStreamImage := ctrl.getMostRecentImageTagForImageStream(checkImageStream, "latest")
+			if mostRecentStreamImage != nil {
+
+				if mostRecentPoolImage.Created.Before(&mostRecentStreamImage.Created) {
```

Comment:
Would it be possible to tag with the rendered config and see if an image has been built already rather than using time?